src/native_background.ts

```diff
@@ -270,14 +275,34 @@ export async function getenv(variable: string) {
  You'll get both firefox binary (not necessarily an absolute path) and flags */
 export async function ffargs(): Promise<string[]> {
     // Using ' and + rather that ` because we don't want newlines
-    let output = await pyeval(
-        'handleMessage({"cmd": "run", ' +
-            '"command": "ps -p " + str(os.getppid()) + " -oargs="})["content"]',
-    )
-    return output.content.trim().split(" ")
+    if ((await browserBg.runtime.getPlatformInfo()).os === "win") {
+        return []
+    } else {
+        let output = await pyeval(
+            'handleMessage({"cmd": "run", ' +
+                '"command": "ps -p " + str(os.getppid()) + " -oargs="})["content"]',
+        )
+        return output.content.trim().split(" ")
+    }
 }
 
 export async function getProfileDir() {
+    // Windows users must specify their Firefox profile directory
+    // via 'set profiledir [directory]'. Windows profile directory
+    // path must be properly escaped. For example, a correct way
+    // to set 'profiledir' on Windows is as shown below:
+    //
+    // : set profiledir C:\\Users\\<User-Name>\\AppData\\Roaming\\Mozilla\\Firefox\\Profiles\\8s21wzbh.Default
+    if ((await browserBg.runtime.getPlatformInfo()).os === "win") {
+        let win_profiledir = config.get("profiledir")
+        win_profiledir = win_profiledir.trim()
+        if (win_profiledir.length > 0) {
+            return win_profiledir
+        } else {
+            throw new Error("Profile directory not set.")
+        }
```

Comment:
Somewhere we had a more useful error message telling people go to set their profile and they they could find it on about:support, we should do something similar here.

Reply:
[Done!][0]

[0]: https://github.com/gsbabil/tridactyl/commit/ef788a393f09c17858e10e42f61eb00f6a7ed96e#diff-9ddd9172e4a54cdd30d56e5f10d5facaR315